extensions/pkg/controller/worker/genericactuator/actuator_reconcile.go

```diff
@@ -309,6 +316,13 @@ func (a *genericActuator) deployMachineDeployments(ctx context.Context, cluster
 								Taints: deployment.Taints,
 							},
 						},
+						MachineControllerConfig: &machinev1alpha1.MachineControllerConfig{
+							MachineDrainTimeout:    workerPool.Settings.MachineDrainTimeout,
+							MachineCreationTimeout: workerPool.Settings.MachineCreationTimeout,
+							MachineHealthTimeout:   workerPool.Settings.MachineHealthTimeout,
+							MaxEvictRetries:        workerPool.Settings.MaxEvictRetries,
+							NodeConditions:         workerPool.Settings.NodeConditions,
```

Comment:
Nil pointer danger? workerPool can be nil and also Settings can be nil

Reply:
Yes, good catch, seemingly it is one. Out of curiosity, I tested a couple of times now- removing settings, but it doesn't throw the nil-pointer, couldn't understand why. I also found that, I had mistakenly missed the `omitempty [here](https://github.com/gardener/gardener/blob/42cf17c3d3269b329894d30141f30b32f5202d22/pkg/apis/core/v1alpha1/types_shoot.go#L923). I fixed, that locally, and tried still couldn't produce the nil-pointer. 

Anyways, I will try to align the code to check if corresponding fields[drainTimeout, etc] are set and then only use here. Thanks for pointing out.

Reply:
This code-path is removed based on this comment: https://github.com/gardener/gardener/pull/2563#discussion_r457007752 .